auto-cancel.md

```diff
@@ -0,0 +1,164 @@
+
+
+- [Auto-cancel queued pipelines on a new push](#auto-cancel-queued-pipelines-on-a-new-push)
+- [Auto-cancel both running and queued pipelines on a new push](#auto-cancel-both-running-and-queued-pipelines-on-a-new-push)
+- [Auto-cancel all pipelines on a new push to a non-master branch](#auto-cancel-all pipelines-on-a-new-push-to-a-non-master-branch)
+- [See aslo](#see-also)
+
+## Auto-cancel queued pipelines on a new push
+
+If your Git commit log would look like this:
+
+``` txt
+- commit C <- queued
+- commit B <- queued
+- commit A <- a workflow / pipeline is currently running for this revision
+```
+then when you push new commit D with `.semaphore/semaphore.yml` updated in this fashion:
+
+``` yaml
+version: "v1.0"
+name: Build and test pipeline
+agent:
+  machine:
+    type: e1-standard-2
+    os_image: ubuntu1804
+
+auto_cancel:
+  queued:
+    when: "true" # enable auto-cancel for branches, tags and pull-requests
+
+blocks:
+  - name: Unit Tests
+    task:
+      jobs:
+      - name: "Unit Tests"
+        commands:
+          - echo Testing...
+```
+
+the following will happen:
+
+``` txt
+- commit D <- queued, waiting for the pipeline from commit A to finish
+- commit C <- canceled
+- commit B <- canceled
+- commit A <- pipeline running for this revision
+```
+
+The pipeline from the newest commit D will arrive on top of the queue by canceling
+all other pipelines previously in the queue that are not yet started to run, which
+are in this case pipelines from commits B and C.
+
+The currently running pipeline from commit A will not be affected and will be
+allowed to finish.
+This will make sure that no execution time is wasted without providing you any
+feedback.
+
+## Auto-cancel both running and queued pipelines on a new push
+
+Let's say your Git commit log looks like this:
+
+``` txt
+- commit C <- queued
+- commit B <- queued
+- commit A <- a workflow / pipeline is currently running for this revision
+```
+If you push new commit D with `.semaphore/semaphore.yml` updated in this fashion:
+
+``` yaml
+version: "v1.0"
+name: Build and test pipeline
+agent:
+  machine:
+    type: e1-standard-2
+    os_image: ubuntu1804
+
+auto_cancel:
+  running:
+    when: "true"
+
+blocks:
+  - name: Unit Tests
+    task:
+      jobs:
+      - name: "Unit Tests"
+        commands:
+          - echo Testing...
+```
+
+Then the following will happen:
+
+``` txt
+- commit D <- pipeline running for newest revision
+- commit C <- canceled
+- commit B <- canceled
+- commit A <- pipeline stopped
+```
+
+The execution of pipeline from commit A will be stopped and pipelines initiated
+from commits B and C that were waiting in the queue will be canceled.
+
+This will allow pipeline from newest commit D to start executing and provide you
+with feedback as soon as possible after a push.
+
+## Auto-cancel all pipelines on a new push to a non-master branch
+
+Often the auto-cancel strategy is ideal for feature branches during development.
```

Comment:
I would drop the `Ofter the` part of the sentence. It is ok to be prescriptive in use cases.